lib/core/realtime/hotelClerk.js

```diff
@@ -109,20 +114,104 @@ class HotelClerk {
     this.customers = new Map();
   }
 
+  async init () {
+    /**
+     * Create a new, empty room.
+     * @param {string} index
+     * @param {string} collection
+     * @param {string} roomId
+     * @returns {boolean} status indicating if the room was created or not
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:room:new',
+      (index, collection, roomId) => this.newRoom(index, collection, roomId));
+
+    /**
+     * Returns the number of connections listening to a given room
+     *
+     * @param  {string} roomId
+     * @return {number}
+     * @throws {NotFoundError} If the roomId does not exist
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:room:size',
+      roomId => this.getRoomSize(roomId));
+
+
+    /**
+     * Joins an existing room.
+     * @param  {Request} request
+     * @returns {Promise}
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:join', request => this.join(request));
+
+    /**
+     * Return the list of index, collection, rooms (+ their users count)
+     * on all index/collection pairs that the requesting user is allowed to
+     * subscribe
+     *
+     * @param  {string} roomId
+     * @return {number}
+     * @throws {NotFoundError} If the roomId does not exist
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:list',
+      requestContext => this.list(requestContext));
+
+    /**
+     * Given an index, returns the list of collections having subscriptions
+     * on them.
+     * @param  {string} index
+     * @return {Array.<string>}
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:listCollections',
+      index => this.listCollections(index));
+
+    /**
+     * Removes a user and all their subscriptions.
+     * @param  {RequestContext} requestContext
+     */
+    this.kuzzle.onAsk(
+      'core:realtime:disconnect',
```

Comment:
I'd rather call this `core:realtime:unsubscribeAll`, since `disconnect` would imply its connection is closed, which is not.

Reply:
resolved in #1735: this event has been renamed `core:realtime:user:remove`